YellowNotes.Api/Filters/ValidateTokenFilter.cs

```diff
@@ -0,0 +1,24 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using YellowNotes.Api.Extensions;
+using YellowNotes.Core.Utility;
+
+namespace YellowNotes.Api.Filters
+{
+    public class ValidateTokenFilter : ActionFilterAttribute
+    {
+        public override void OnActionExecuting(ActionExecutingContext context)
+        {
+            var userEmail = context.HttpContext.GetEmailFromClaims();
+
+            var controller = context.Controller as ControllerBase;
```

Comment:
I'm not sure right now, I'm giving 80% that `context` has `Request.Headers`, we don't need `ControllerBase` here